### PR TITLE
Add DataAPI.invrefpool

### PR DESCRIPTION
**Pull request by bkamins**

### CODE REVIEW

File: src/array.jl [hunk 1]
```diff
@@ -1097,9 +1097,31 @@ Base.LinearIndices(x::CategoricalRefPool) = axes(x, 1)
 DataAPI.refarray(A::CatArrOrSub) = refs(A)
 DataAPI.refpool(A::CatArrOrSub{T}) where {T} =
     CategoricalRefPool{eltype(A), typeof(pool(A))}(pool(A))
+DataAPI.invrefpool(A::CatArrOrSub{T}) where {T} =
+    CategoricalInvRefPool{eltype(A), typeof(pool(A).invpool)}(pool(A).invpool)
```
Comment (bkamins) [inline src/array.jl:1101]:
```suggestion
    CategoricalInvRefPool{eltype(A), typeof(pool(A).invindex)}(pool(A).invindex)
```

File: test/13_arraycommon.jl [hunk 2]
```diff
@@ -2165,6 +2165,11 @@ end
         @test_throws BoundsError rp[-1]
         @test_throws BoundsError rp[end + 1]
         @test_throws MethodError similar(rp)
+
+        irp = DataAPI.invrefpool(y)
+        for lev in eltype(y) >: Missing ? [missing, levels(y)] : levels(y)
```
Comment (bkamins) [inline test/13_arraycommon.jl:2170]:
```suggestion
        for lev in (eltype(y) >: Missing ? [missing, levels(y)] : levels(y))
```

File: Project.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name = "CategoricalArrays"
 uuid = "324d7699-5711-5eae-9e2f-1d82baa6b597"
-version = "0.9.0"
+version = "0.10.0"
```
Comment (nalimilan) [inline Project.toml:3]: Better release 0.9.1 with this change as it's not breaking. We can move to 0.10 later if needed.

File: test/13_arraycommon.jl [hunk 4]
```diff
@@ -2165,6 +2165,17 @@ end
...
         @test_throws BoundsError rp[end + 1]
         @test_throws MethodError similar(rp)
+
+        irp = DataAPI.invrefpool(y)
+        for lev in (eltype(y) >: Missing ? [missing; levels(y)] : levels(y))
+            @test isequal(rp[irp[lev]], lev)
+        end
+
+        @test_throws KeyError irp[1]
+        @test_throws KeyError irp[z]
```
Comment (nalimilan) [inline test/13_arraycommon.jl:2175]: Where is `z` defined?

Comment (bkamins) [reply test/13_arraycommon.jl:2175]: it is not - I do not know why it is not `"z"` as in my local file :confused:

Comment (bkamins) [inline test/13_arraycommon.jl:2175]:
```suggestion
        @test_throws KeyError irp["z"]
```

File: Project.toml (hunk 3 above)
Comment (bkamins) [reply Project.toml:3]: done

Comment (bkamins) [inline Project.toml:3]:
```suggestion
version = "0.9.1"
```

**Comment by bkamins** [comment]
@nalimilan - the CI failures seem to be unrelated (something is problematic with `show` on 1.0 and on nightly), but you know the code base better. A quick look at it would be appreciated. Thank you!

File: test/13_arraycommon.jl [hunk 5]
```diff
@@ -1638,8 +1638,8 @@ end
         @test sprint((io,a)->show(io, "text/plain", a), x) ==
             """
             3-element $CategoricalArray{$(Union{Missing,Int}),1,UInt32}:
-             2      
-             1      
+             2
+             1
```
Comment (nalimilan) [inline test/13_arraycommon.jl:1642]: This is the reason for the 1.0 failure. :-D
```suggestion
             2      
             1      
```

**Comment by bkamins** [comment]
Thank you!

**Comment by nalimilan** [comment]
https://github.com/JuliaRegistries/General/pull/29281

**Pull request merged**